### PR TITLE
Use stream encoding instead of locale preferred encoding

### LINKED ISSUE #53: Display coding problem
Code：
# -*- coding:utf-8 -*-
import better_exceptions
import sys
print(sys.stdout.encoding)
better_exceptions.hook()

foo = 52

def shallow(a, b):
    deep(a + b)

def deep(val):
    global foo
    assert val > 10 and foo == 60

bar = foo - 50
shallow(bar, 15)
shallow(bar, 2)

display:
"C:\Program Files\Python36\python.exe" C:/Users/Alex/PycharmProjects/Test/Test02.py
UTF-8
Traceback (most recent call last):
  File "C:/Users/Alex/PycharmProjects/Test/Test02.py", line 20, in <module>
    shallow(bar, 15)
    ��       �� 2
    �� <function shallow at 0x000001860AAEF730>
  File "C:/Users/Alex/PycharmProjects/Test/Test02.py", line 11, in shallow
    deep(a + b)
    ��    ��   �� 15
    ��    �� 2
    �� <function deep at 0x000001860B079620>
  File "C:/Users/Alex/PycharmProjects/Test/Test02.py", line 16, in deep
    assert val > 10 and foo == 60
           ��            �� 52
           �� 17
AssertionError: assert val > 10 and foo == 60

Process finished with exit code 1

system: windows 10
python: 3.6

### PR DESCRIPTION
**Pull request by Delgan**
Hi.

This is intended to fix #53, finally. :)

As discussed, I replaced `locale.getpreferredencoding()` with `STREAM.encoding`. Just in case, I added a fallback to `ascii` if `STREAM` has no or invalid `encoding` attribute. For the tests, the encoding can be configured with the [`PYTHONIOENCODING`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONIOENCODING) environment variable in place of `LANG` and `LC_ALL`. Also, I had to remove `encoding.py` and include it directly into `color.py` to avoid circular dependency.

Hopefully, this should solve reported encoding issues. ;)

### CODE REVIEW

File: better_exceptions/color.py [hunk 1]
```diff
@@ -6,20 +6,45 @@
...
 import os
 import struct
 import sys
 
 from .context import PY3
-from .encoding import to_byte as _byte
 
 
 STREAM = sys.stderr
+ENCODING = getattr(STREAM, "encoding", None) or "ascii"
```
Comment (Qix-) [inline better_exceptions/color.py]: Better to default to utf-8 IMO.

**Comment by Qix-** [comment]
@delgan does this supercede #69?

File: better_exceptions/color.py (hunk 1 above)
Comment (Delgan) [reply better_exceptions/color.py]: @Qix- Yeah, I think you are right, that would avoid possible `UnicodeEncodeError`.

**Comment by Delgan** [comment]
> @Delgan does this supercede #69?

@Qix- There is definitely overlaps between #88 and #69, but #69 tried to go further by removing a bunch of hacky functions used for Python 2 compatibility with `colorama`.
Re-reading what I commented on the PR at that time, it seems this was mostly personal convictions about how encoding should be done. This does not solve any actual issue, so it's ok to merge #88 and reject #69 (I could simply re-implement the changes in the future if needed).

**Comment by Qix-** [comment]
Sounds good :)

**Comment by danyeaw** [comment]
PEP597 Encoding Warnings gives warnings about locale preferred encoding if you have the warnings enabled. What needs to happen to get this PR merged?

**Comment by Qix-** [comment]
Just a rebase on top of current master. I'm out of the country at the moment so my capacity for code changes is low.

**Comment by Delgan** [comment]
Hey there. :) 

I just rebased the PR. :+1:
It's quite old, but I first glance, it seems the proposed workaround is still relevant.

**Review by Qix- — APPROVED** [review]

**Comment by Qix-** [comment]
It's going to be a little bit before I can push a release, sorry in advance. @Delgan if you're interested in being added as a pypi maintainer just let me know what your pypi username is.

**Comment by Delgan** [comment]
@Qix- Thank you for your trust, but I honestly prefer not to handle the administrative part. Let me know if you need help with other developments, though. I would be happy to contribute. :+1:

**Pull request merged**